File: 0247-Strobogrammatic Number II/0247-Strobogrammatic Number II.cpp
class Solution {
public:
    vector<string> findStrobogrammatic(int n) {
        vector<string> result;
        if (n <= 0)
        {
            return result;
        }
        
        if (n % 2)
        {
            --n;
            result = {"0", "1", "8"};
        }
        else
        {
            result = {""};
        }
        
        while (n)
        {
            vector<string> temp;
            for (string s : result)
            {
                if (n != 2)
                {
                    temp.push_back("0" + s + "0");
                }
                temp.push_back("1" + s + "1");                
                temp.push_back("6" + s + "9");
                temp.push_back("8" + s + "8");
                temp.push_back("9" + s + "6");
            }
            
            n -= 2;
            result = temp;
        }
        
        return result;
    }
};

File: 0201-0300/0247-Strobogrammatic Number II/0247-Strobogrammatic Number II.py
class Solution:
    def findStrobogrammatic(self, n: int) -> List[str]:
        if n <= 0:
            return []
        if n & 1:
            n -= 1
            result = ['0', '1', '8']
        else:
            result = ['']
        
        while n:
            temp = []
            for s in result:
                if n != 2:
                    temp.append('0' + s + '0')
                temp.append('1' + s + '1')
                temp.append('6' + s + '9')
                temp.append('8' + s + '8')
                temp.append('9' + s + '6')
            n -= 2
            result = temp
        return result
